metacatalog/ext/io/importer.py

```diff
@@ -1,81 +1,137 @@
 import os
 
 import pandas as pd
+import sqlalchemy as sa
 from sqlalchemy.orm import object_session
+from sqlalchemy.dialects.postgresql import ARRAY
 
 from metacatalog.models.entry import Entry
 from metacatalog.models.timeseries import TimeseriesPoint
 
 
-def import_to_internal_table(entry, datasource, data, mapping=None, **kwargs):
+def import_to_internal_table(entry, datasource, data, force_data_names=False, **kwargs):
     """Import to internal DB
 
-    The given data is imported into the table 
-    as specified in the datasource. The data column names need to
-    fit the names as implemented in the database. The mapping 
-    keyword can be used to rename
+    The given data is imported into the table
+    as specified in the datasource.
+    If force_data_names=True the column names of the imported data are saved in
+    the datasource, otherwise the standard column names in
+    entry.variable.column_names are used. The column names in
+    datasource.data_names are used when exporting the data.
     """
     # check that entry is valid
     assert Entry.is_valid(entry)
-    
+
     if isinstance(data, pd.Series):
         data = pd.DataFrame(data)
 
     # reset the index
     imp = data.reset_index(level=0, inplace=False)
-    
-    # rename if mapping is given
-    if mapping is not None and isinstance(mapping, dict):
-        imp.rename(columns=mapping, inplace=True)
-
-    # set entry_id
-    if 'entry_id' not in imp.columns:
-        imp['entry_id'] = entry.id
 
     # check if a session was passed
     if 'session' in kwargs.keys():
         session = kwargs['session']
     else:
         session = object_session(entry)
 
+    # get index, drop it afterwards
+    if 'index' in imp.columns:
+        index = imp.index
+        imp.drop('index', axis=1, inplace=True)
+    elif 'tstamp' in imp.columns:
+        index = imp.tstamp
+        imp.drop('tstamp', axis=1, inplace=True)
+
+    # get the column names - exclude everthing that stores precisions
+    data_columns = [col for col in imp.columns.tolist() if not col.startswith('precision')]
+
+    # get the precision columns
+    precision_columns = [col for col in imp.columns.tolist() if col.startswith('precision')]

```

Comment:
This only works if the user names the precision columns `precision1`, `precision2` and so on. If the user names the columns for example `prec1`, `prec2`, it won´t work (in this case, `force_data_names` will become `True` and ALL the columns will be named like in the uploaded data). 

If the user would use column names like `prec1` or `uncertainty`, these columns won´t be stored in the `precision` column of timeseries_array, but in the data column.

In the documentation we could write that the user has to name precision columns `precision1`, `precision2` and so on, but even this is not that good, because the information is lost which precision metric is used (which is also the case now).

What do you think about that?

Reply:
@mmaelicke

Reply:
I guess most users won't upload precision data anyway. The maybe better approach would be to ask for data and precision separatedly, like:

```Python
def import_to_internal_table(entry, datasource, data, precision=None, force_data_names=False, **kwargs):
```

Then in (maybe a second PR) we can check precision. If it's a array of same length, we store it, If ints and integer, we assume an absolute error, if it's a float 0 < precision < 1 we assume a relative error. That's something we can discuss later.
Column Names for precision can also be added to the datasouce (defaulting to `'precision'` only, if more is given, we always mimic the force names behavior). The numeric stuff doesn't need to be implemented yet.